### DISCUSSION TITLE
AKS - KEDA - ScaledObject - ServiceBus Queue - Message Count Triggered

### DISCUSSION THREAD

**Discussion by atulthummar** [kedacore/keda#1837] [Q&A / Need Help]
@tomkerkhove Hello Guys,

I have question on AKS - KEDA - ScaledObject - ServiceBus Queue - Message Count Trigger related
I have configured as like below 
apiVersion: keda.sh/v1alpha1 
kind: ScaledObject
metadata:
  name: azure-function-scaler
spec:
  scaleTargetRef:
    name: azure-function
  cooldownPeriod:  5  
  minReplicaCount: 0 
  maxReplicaCount: 100
  triggers:
  - type: azure-servicebus
    metadata:
      queueName: queue
      connectionFromEnv: SERVICEBUS_CONNECTIONSTRING
      namespace: testnamespace
      **messageCount: '10'**
    authenticationRef:
     name: azure-function-trigger-auth

Now my question is How KEDA created PODs on based on Message count
1. Does it wait until  reach to 10 then create POD ?
2. If Bus Queue has only 1 Messages or less then 10 message what happen ?
3.  How we should combine POD trigger based on Message Count and CPU usage multiple condition ? is it possible

**Comment by zroubalik** [comment]
1. yes
2. yes
3. you can define both in the ScaledObject (you can use multiple triggers), the greater value will take the precedence and the final replica count will be based on that.

**Reply by atulthummar** [reply]
#1 Are you sure ? - in my case if I triggered one message then also it get run on single POD
#2 Also it get executed on Single POD it does not wait till message count reach 10
#3 Thanks for confirmation 

Is there any way to check which Message is running on which POD ?

**Comment by atulthummar** [comment]
@jeffhollan @tomkerkhove Any input will be highly appreciated Thanks!
